### PR TITLE
Fixes mapbox popups with non-numeric values

### PR DESCRIPTION
**Pull request by glynnbird**
If you choose a non-numeric value in a MapBox view, it doesn't get displayed. This pull-request fixes that.

See issue #203

### CODE REVIEW

File: pixiedust/display/chart/renderers/baseChartDisplay.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def validateOptions(self):
             del self.options[key]
 
     def getExtraFields(self):
-        return []
+        return self.getNonNumericValueFields()
```
Comment (DTAIEB) [inline pixiedust/display/chart/renderers/baseChartDisplay.py]: @glynnbird baseChartDisplay is used by every single renderer, not just mapbox. Adding non numeric value fields as extra fields all the time will certainly have side effects. Why is that needed here?

Comment (glynnbird) [reply pixiedust/display/chart/renderers/baseChartDisplay.py]: It's like that because this call is made to get a dataframe prior to rendering:

https://github.com/ibm-watson-data-lab/pixiedust/blob/issue203/pixiedust/display/chart/renderers/mapbox/mapBoxMapDisplay.py#L81

Without the extraFields change, then the non-numeric fields were not present in the Pandas data frame and couldn't be rendered.

Comment (DTAIEB) [reply pixiedust/display/chart/renderers/baseChartDisplay.py]: I see, then the correct usage is to override this method in the Mapbox concrete class.

File: pixiedust/display/chart/renderers/baseChartDisplay.py [hunk 2]
```diff
@@ -114,6 +114,7 @@ def wrapper(cls, *args, **kwargs):
 class BaseChartDisplay(with_metaclass(ABCMeta, ChartDisplay)):
 
     commonOptions = co
+    extraFields = []
```
Comment (DTAIEB) [inline pixiedust/display/chart/renderers/baseChartDisplay.py]: @glynnbird I'm afraid we still have the same issue. As declared extraFields is a static field that once modified will apply to all subclasses.
You should leave the default body of getExtraFields function to return [] and have it overridden in the concrete class to return the non-numeric fields

Comment (glynnbird) [reply pixiedust/display/chart/renderers/baseChartDisplay.py]: Gotcha. New commit on its way
